plugin/evm/wrapped_block.go

```diff
@@ -441,15 +440,14 @@ func (b *wrappedBlock) verifyPredicates(predicateContext *precompileconfig.Predi
 		return nil
 	}
 
-	predicateResults := predicate.BlockResults{}
-	for _, tx := range b.ethBlock.Transactions() {
-		results, err := core.CheckPredicates(rules, predicateContext, tx)
-		if err != nil {
-			return err
-		}
-		predicateResults.Set(tx.Hash(), results)
+	predicateResults, err := core.CheckBlockPredicates(
+		rules,
+		predicateContext,
+		b.ethBlock.Transactions(),
+	)
+	if err != nil {
+		return err
 	}
-	// TODO: document required gas constraints to ensure marshalling predicate results does not error
```

Comment:
Why is this TODO no longer relevant here?

Reply:
It was documented: https://github.com/ava-labs/avalanchego/blob/master/vms/evm/predicate/results.go#L20-L32